packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2445,6 +2471,11 @@ export class ContainerRuntime
 	}
 
 	public setConnectionState(connected: boolean, clientId?: string) {
+		// Validate we have consistent state
+		const currentClientId = this._audience.getSelf()?.clientId;
+		assert(clientId === currentClientId, "same clientId");
+		assert(this.clientId === currentClientId, "same clientId");
```

Comment:
You should make the assert messages different so that if it hits, which one was hit.

Reply:
Also the message should maybe say "different clientIds" instead because that's the error right.